internal/setup/setup_test.go

```diff
@@ -1063,6 +1063,197 @@ var setupTests = []setupTest{{
 		`,
 	},
 	relerror: `invalid slice definition filename: "a.yaml"`,
+}, {
+	summary: "Package essentials with same package slice",
+	input: map[string]string{
+		"slices/mydir/mypkg.yaml": `
+			package: mypkg
+			essential:
+				- mypkg_essential
+			slices:
+				slice1:
+					contents:
+						/file:
+						/foo/bar/baz:
+				slice2:
+					contents:
+						/d**:
+				essential:
+					contents:
+						/essential-file:
+		`,
+	},
+	release: &setup.Release{
+		DefaultArchive: "ubuntu",
+
+		Archives: map[string]*setup.Archive{
+			"ubuntu": {
+				Name:       "ubuntu",
+				Version:    "22.04",
+				Suites:     []string{"jammy"},
+				Components: []string{"main", "universe"},
+				PubKeys:    []*packet.PublicKey{testKey.PubKey},
+			},
+		},
+		Packages: map[string]*setup.Package{
+			"mypkg": {
+				Archive: "ubuntu",
+				Name:    "mypkg",
+				Path:    "slices/mydir/mypkg.yaml",
+				Slices: map[string]*setup.Slice{
+					"slice1": {
+						Package: "mypkg",
+						Name:    "slice1",
+						Essential: []setup.SliceKey{
+							{"mypkg", "essential"},
+						},
+						Contents: map[string]setup.PathInfo{
+							"/file":        {Kind: "copy"},
+							"/foo/bar/baz": {Kind: "copy"},
+						},
+					},
+					"slice2": {
+						Package: "mypkg",
+						Name:    "slice2",
+						Essential: []setup.SliceKey{
+							{"mypkg", "essential"},
+						},
+						Contents: map[string]setup.PathInfo{
+							"/d**": {Kind: "glob"},
+						},
+					},
+					"essential": {
+						Package: "mypkg",
+						Name:    "essential",
+						Essential: []setup.SliceKey{
+							{"mypkg", "essential"},
```

Comment:
This one is awkward. We probably don't want self-references like this?  This should probably be an error if explicit, and if implicit via the package-level essentials, we should drop it from this particular slice.

Reply:
Changed, it is now an error if a user adds the slice to the slice-level essentials. For package-level essentials, we just dont add the slice to itself.